libp2p/Host.cpp

```diff
@@ -355,20 +312,42 @@ void Host::runAcceptor()
 				}
 			}
 			
-			if (!success && m_socket->is_open())
+			if (!success && s->is_open())
 			{
 				boost::system::error_code ec;
-				m_socket->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
-				m_socket->close();
+				s->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
+				s->close();
 			}
 
 			m_accepting = false;
+			delete s;
```

Comment:
Did you mean destructor? I need to look into this. I think that 'moved' pointers can still be deleted. There may also be a proper way to do this with unique_ptr.